### ISSUE TITLE
API for listing file IDs using only a download token

### ISSUE THREAD

**Issue by multimeric** [filesender/filesender#1555]
I'm trying to automate the download process using the API. The issue is that, if you have a download URL such as `/?s=download&token=2d335494-b624-5942-a152-a5579eeccff2`, there is no way to programatically list all of the files that are available. 

Once you have the list of file IDs, you can easily use `/download.php?token=2d335494-b624-5942-a152-a5579eeccff2&files_ids=ABC,DEF,GHI` to download the files programmatically.

Therefore, it would be great if there were a REST API that did this. The authentication should be similar to `download.php` in that it doesn't require authentication but does require a download token. The endpoint would simply return a JSON list of file IDs.

**Comment by monkeyiq** [comment]
This is almost there. At the moment if you go a GET for a transfer id `$tid` on
```
https://localhost/filesender/rest.php/transfer/$tid
```
then you can see all the IDs. It is probable that a bridge to get the transfer id $tid from the token is needed. Or that REST api expanded to allow $token -> file id list.

The below pull request allows the fileids to be obtained making a REST GET with `.../rest.php/transfer/fileids?token=2d335494-b624-5942-a152-a5579eeccff2`

https://github.com/filesender/filesender/pull/1569

**Comment by multimeric** [comment]
Yeah that was part of the issue. As a download user with a download link, all you have is a download token, but this doesn't authenticate you to list transfers and therefore get the file IDs. The *sender* of the files will have no problem, but the recipient will.

**Comment by multimeric** [comment]
Just noting here that it would also be nice to get the full file name from this API, because currently there's no way to recreate the full filename with its directory names just via the `/download.php` endpoint.

**Comment by madsi1m** [comment]
@multimeric, perhaps you would like to create a pull request for consideration?

**Comment by monkeyiq** [comment]
I think this is there but remains behind a wall where you have to be user authenticated. The user authentication is not really needed if you have the token as that will mirror the security of the web interface. I think perhaps the user authentication was put in there to err on the side of caution. But this access control discussion is secondary to the information offered.

In particular this call will give all metadata about a file including it's name. If that name does not include the path then I am happy to include that in this API call as well.
https://github.com/filesender/filesender/blob/f2a2f9570e9129d600739c39cdc7eb256ea6020c/classes/rest/endpoints/RestEndpointTransfer.class.php#L202C1-L202C15

**Comment by multimeric** [comment]
> I think this is there but remains behind a wall where you have to be user authenticated. The user authentication is not really needed if you have the token as that will mirror the security of the web interface.

Yes, this is exactly my issue! The recipient doesn't have authentication.

> In particular this call will give all metadata about a file including it's name.

I tried playing around with this API but I couldn't get it to work. Is it `/transfer/fileids?token=XXX`? If so, that doesn't work for my use case because it enforces authentication:
https://github.com/filesender/filesender/blob/f2a2f9570e9129d600739c39cdc7eb256ea6020c/classes/rest/endpoints/RestEndpointTransfer.class.php#L187-L189

**Comment by monkeyiq** [comment]
I will take a look at the authentication. I think in this case I erred on the side of caution. The token should be the main thing that is hard to guess. This might be loosened to only doing the auth check if the option is selected for the transfer that the recipient needs to be logged in in order to download.

**Comment by adminnr1** [comment]
We have seen [1569](https://github.com, but this is no solution for us. 
Desired solution is to download files via API without authentication. 
This setting should be configurable by the FileSender admin.
